src/tests/unittests/validation/DepthStencilStateValidationTest.cpp

```diff
@@ -0,0 +1,82 @@
+        .SetStencilMask(nxt::Face::Both, 0x1)
+        .GetResult();
+
+    // Success for empty builder
+    nxt::DepthStencilState buf1 = AssertWillBeSuccess(device.CreateDepthStencilStateBuilder())
+        .GetResult();
+
+    // Test failure when specifying properties multiple times
+    nxt::DepthStencilState buf2 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetDepthWriteEnabled(true)
+        .SetDepthWriteEnabled(false)
+        .GetResult();
+
+    nxt::DepthStencilState buf3 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetDepthCompareFunction(nxt::CompareFunction::Less)
+        .SetDepthCompareFunction(nxt::CompareFunction::Greater)
+        .GetResult();
+
+    // Test success when setting properties on separate faces
+    nxt::DepthStencilState buf4 = AssertWillBeSuccess(device.CreateDepthStencilStateBuilder())
+        .SetStencilMask(nxt::Face::Front, 0x00)
+        .SetStencilMask(nxt::Face::Back, 0xff)
+        .GetResult();
+
+    nxt::DepthStencilState buf5 = AssertWillBeSuccess(device.CreateDepthStencilStateBuilder())
+        .SetStencilFunction(nxt::Face::Front, nxt::CompareFunction::Less,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .SetStencilFunction(nxt::Face::Back, nxt::CompareFunction::Greater,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .GetResult();
+
+    // Test failure when setting properties on a face multiple times
+    nxt::DepthStencilState buf6 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetStencilMask(nxt::Face::Back, 0x00)
+        .SetStencilMask(nxt::Face::Back, 0xff)
+        .GetResult();
+
+    nxt::DepthStencilState buf7 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetStencilFunction(nxt::Face::Back, nxt::CompareFunction::Less,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .SetStencilFunction(nxt::Face::Back, nxt::CompareFunction::Greater,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .GetResult();
+
+    nxt::DepthStencilState buf8 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetStencilMask(nxt::Face::Both, 0x00)
+        .SetStencilMask(nxt::Face::Back, 0xff)
+        .GetResult();
+
+    nxt::DepthStencilState buf9 = AssertWillBeError(device.CreateDepthStencilStateBuilder())
+        .SetStencilFunction(nxt::Face::Both, nxt::CompareFunction::Less,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .SetStencilFunction(nxt::Face::Back, nxt::CompareFunction::Greater,
+            nxt::StencilOperation::Replace, nxt::StencilOperation::Replace, nxt::StencilOperation::Replace)
+        .GetResult();
+}
```

Comment:
We also need to add a test for setting a depth stencil test on a compute pipeline. This is significant amount of work so feel free to do it in a follow up PR. You'll need to add a function like Utils.h CreateShaderModule to the tests and also add a dummy compute shader.